### ISSUE TITLE
add tf.keras.layers.InputSpec to custom layers layers

### ISSUE THREAD

**Issue by sascha-kirch** [sascha-kirch/DeepSaki#13]
with tf.keras.layers.InputSpec(dim=xxx) one can specify the specification for an input when the layer is called. 

Dont forget: unit test

**Comment by sascha-kirch** [comment]
https://www.tensorflow.org/api_docs/python/tf/keras/layers/InputSpec. I could also add dtypey, especially for the Fourier layers that expect complex inputs.
